RealmSwift/Impl/ObjcBridgeable.swift

```diff
@@ -0,0 +1,180 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Foundation
+import Realm
+
+/// A type which can be bridged to and from Objective-C.
+///
+/// Do not use this protocol or the functions it adds directly.
+public protocol _ObjcBridgeable {
```

Comment:
Why is this public? I know that the public protocols that use cannot refine the protocol if it were internal, but why not have the final types implement it?

Reply:
Ah, the final types can be custom user types. Disregard.

Reply:
We also can't use it as a protocol requirement on public functions without it being public, which lead to a bunch of awful dynamic casts which would always succeed.